processor/attributesprocessor/attributes.go

```diff
@@ -121,36 +127,66 @@ func (a *attributesProcessor) processSpan(span pdata.Span, serviceName string) {
 		case DELETE:
 			attrs.Delete(action.Key)
 		case INSERT:
-			av, found := getSourceAttributeValue(action, attrs)
+			av, found := getSourceAttributeValues(action, attrs)
 			if !found {
 				continue
 			}
-			attrs.Insert(action.Key, av)
+			for key, value := range av {
+				attrs.Insert(key, value)
+			}
 		case UPDATE:
-			av, found := getSourceAttributeValue(action, attrs)
+			av, found := getSourceAttributeValues(action, attrs)
 			if !found {
 				continue
 			}
-			attrs.Update(action.Key, av)
+			for key, value := range av {
+				attrs.Update(key, value)
+			}
 		case UPSERT:
-			av, found := getSourceAttributeValue(action, attrs)
+			av, found := getSourceAttributeValues(action, attrs)
 			if !found {
 				continue
 			}
-			attrs.Upsert(action.Key, av)
+			for key, value := range av {
+				attrs.Upsert(key, value)
+			}
 		case HASH:
 			hashAttribute(action, attrs)
 		}
 	}
 }
 
-func getSourceAttributeValue(action attributeAction, attrs pdata.AttributeMap) (pdata.AttributeValue, bool) {
+func getSourceAttributeValues(action attributeAction, attrs pdata.AttributeMap) (map[string]pdata.AttributeValue, bool) {
```

Comment:
I'm not super familiar with Go internals, but I would expect it would be more efficient to return an array of key, value pairs here instead of a map. Maybe its even worth optimising for the non-regex case and having a separate code path

Reply:
I changed it to always use an array, I think the overhead of using the array should be negligible for the non regex case and I think this is cleaner than creating a separate code path. What do you think?